### ISSUE TITLE
Make 2.11.2 more generic to also apply to in exemple JWTs (not just SAML assertions)

### ISSUE THREAD

**Issue by TobiasAhnoff** [OWASP/ASVS#2325]
Current 2.11.2 is

> Verify that the presence and integrity of digital signatures on SAML assertions is validated, rejecting any assertions that are unsigned or have invalid signatures.

A suggestion is to make this more generic, since this also applies for JWTs, in example when used as OIDC ID Tokens.

> Verify that the presence and integrity of digital signatures for example JWTs or SAML assertions are always validated, and that the request is rejected if this fails (either since it was unsigned or had an invalid signature).

If made more generic, it might be moved to V2.9 as signature validation is important for all authentication scenarios, not just federated authentication. Then it could be rewritten like this:

> Verify that the presence and integrity of digital signatures are always validated, and that the request is rejected if this fails (either since it was unsigned, used a non-approved signature algorithm or had an invalid signature).

**Comment by ImanSharaf** [comment]
It makes sense

**Comment by jmanico** [comment]
I agree, this makes sense

**Comment by tghosth** [comment]
I would like to keep it scoped as authentication assertions and 2.9 is talking about something slightly different which I'd rather now get confused with so I prefer to keep it where it is but would suggest rewording as:

"_Verify that the presence and integrity of digital signatures on authentication assertions (for example on JWTs or SAML assertions) is always validated, rejecting any assertions that are unsigned or have invalid signatures._"

**Comment by tghosth** [comment]
@TobiasAhnoff

**Comment by TobiasAhnoff** [comment]
To me, wording is good, but "Federated" is little misleading...if you have just one "first-party" IDP-service where your client application use OIDC or SAML to authenticate the user (in that IDP), you are not really doing federation with anyone, but it works to keep it in 2.11, maybe change the title to something like  "IDP Authentication"?
